include/buffer_flusher.h

```diff
@@ -0,0 +1,43 @@
+#ifndef BUFFER_FLUSHER_H
+#define BUFFER_FLUSHER_H
+
+#include <mutex>
+#include <condition_variable>
+#include <queue>
+#include <atomic>
+#include <thread>
+
+class GutterTree;
+class flush_struct;
```

Comment:
you declare `flush_struct` as a struct. but since you use it like a class, it probably pays to redeclare the `flush_struct` as a class.

Reply:
I'm not sure I understand the distinction. Can you explain?

Reply:
You're treating `flush_struct`s like `class`es with constructors, fields, and methods. IIRC best practice is for `struct`s to be treated as data objects without too much in-built functionality.